src/support/file.cpp

```diff
@@ -26,7 +26,8 @@ T wasm::read_file(const std::string &filename, Flags::BinaryOption binary, Flags
   if (debug == Flags::Debug) std::cerr << "Loading '" << filename << "'..." << std::endl;
   std::ifstream infile;
   std::ios_base::openmode flags = std::ifstream::in;
-  if (binary == Flags::Binary) flags |= std::ifstream::binary;
+  // asm2wasm.exe built with MinGW would crash with files containing '\r' when opened without the binary flag
```

Comment:
If the code already always handles skipping \r and works on Linux which does not have a non-binary mode, then there's no reason why the file should be opened without the binary flag, and I can't see how it would work better on Visual Studio.

There's other reports of asm2wasm crashing, see for example: https://github.com/WebAssembly/binaryen/issues/327 which crashes at exactly the same spot.